File: Pyvvo.Logistics/ClientApp/src/app/service/Model/variant.ts
import { Status } from "./status";
import { Parameter } from "./parameter";
import { Taxe } from "./taxe";
import { Image } from "./image";
import { Supplier } from "./supplier";
import { Option } from "./Option";
import { Product } from "./product";
import { LinkedVariant } from "./linked-variant";
import { StockLevel } from "./stock-level";

export class Variant {
  id: number;
  createdOn: Date;
  updatedOn: Date;
  referenceNumber: string;
  avgSupplyTime: number;
  lineNumber: number;
  type: string;
  isActive: boolean;
  isTaxable: boolean;
  weight: boolean;
  moq: number;
  initialStockLevel: number;
  initialStockCost: number;
  retailPrice: number;
  specialPrice: number;
  sku: number;
  supplierCode: number;
  ean: string;
  upc: string;
  isbn: string;
  hsCode: string;
  name: string;
  checked: boolean;//Not mapped to .net 
  qte: number;//Not mapped to .net 
  totalPrice: number;//Not mapped to .net 
  status: Status;
  options: Option[];
  parameter: Parameter;
  product: Product;
  taxe: Taxe;
  image: Image;
  supplier: Supplier;
  purchaseOrderLineItems: LinkedVariant[];
  stockLevel: StockLevel[];
}

File: Pyvvo.Logistics.Model/Model/Variant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Pyvvo.Logistics.Model
{
    public class Variant
    {
        [Required, Key] public long Id { get; set; }
        [Required] public long StatusId { get; set; }
        [Required] public long ParameterId { get; set; }
        [Required] public long ProductId { get; set; }
        [Required] public long TaxeId { get; set; }
        [Required] public long ImageId { get; set; }
        [Required] public long SupplierId { get; set; }
        [Required] public DateTime CreatedOn { get; set; }
        public DateTime UpdatedOn { get; set; }
        public string ReferenceNumber { get; set; }
        public double AvgSupplyTime { get; set; }
        public int LineNumber { get; set; }
        public string Type { get; set; }
        public bool IsActive { get; set; }
        public bool IsTaxable { get; set; }
        public string Weight { get; set; }
        public double MOQ { get; set; }
        public double InitialStockLevel { get; set; }
        public double InitialStockCost { get; set; }
        public double RetailPrice { get; set; }
        public double SpecialPrice { get; set; }
        public string SKU { get; set; }
        public string SupplierCode { get; set; }
        public string EAN { get; set; }
        public string UPC { get; set; }
        public string ISBN { get; set; }
        public string HSCode { get; set; }
        public string Name { get; set; }
        public Status Status { get; set; }
        public Parameter Parameter { get; set; }
        public Product Product { get; set; }
        public Taxe Taxe { get; set; }
        public Image Image { get; set; }
        public Supplier Supplier { get; set; }
        public List<Option> Options { get; set; }
        public List<PurchaseOrderLineItem> PurchaseOrderLineItems { get; set; }
        public List<StockLevel> StockLevels { get; set; }
        public List<LinkedVariant> Variants { get; set; }
        public List<LinkedVariant> LinkedVariants { get; set; }
       

    }
}
